obspy/io/sac/util.py

```diff
@@ -462,3 +463,20 @@ def get_sac_reftime(header):
         raise SacHeaderTimeError(msg)
 
     return reftime
+
+
+def _decode_bytes_and_warn(bytestring, key=None, strip_null_terminated=True):
+    """
+    Decodes bytestring to ASCII, ignoring invalid characters if present and
+    issuing a warning if invalid characters are encountered.
+    """
+    if strip_null_terminated:
+        bytestring = re.sub(b'\x00.*', '', bytestring)
```

Comment:
Aye.. trying that one now, looks good from local tests..
The original `.find()` was failing because it sometimes got bytes, now the `re` one is failing because it sometimes gets a str on Py3.. this Py2/3 is still freaking me out sometimes..